lib/widgets/crowd_league_app.dart

```diff
@@ -97,25 +95,35 @@ class _CrowdLeagueAppState extends State<CrowdLeagueApp> {
     return StoreProvider<AppState>(
       store: _store,
       child: StoreConnector<AppState, Settings>(
-          distinct: true,
-          converter: (store) => store.state.settings,
-          builder: (context, settings) {
-            return MaterialApp(
-              navigatorKey: _redux.navKey,
-              navigatorObservers: [NavigationInfoRecorder(_store)],
-              theme: ThemeDataExt.from(settings.lightTheme),
-              darkTheme: ThemeDataExt.from(settings.darkTheme),
-              themeMode: ThemeModeExt.from(settings.brightnessMode),
-              home: AuthOrMain(), // becomes the route named '/'
-              routes: <String, WidgetBuilder>{
-                '/other_auth_options': (context) => OtherAuthOptionsPage(),
-                '/email_auth_options': (context) => EmailAuthOptionsPage(),
-                '/conversation': (context) => MessagesPage(),
-                '/new_conversation': (context) => NewConversationPage(),
-                '/profile': (context) => ProfilePage()
-              },
-            );
-          }),
+        distinct: true,
```

Comment:
There's actually not much changed here - the changes start from `home:` 
- there's an extra StoreConnector that rebuilds on changes to `state.pagesData` 
- the MaterialApp now takes a Navigator, which
  - takes a list of pages, which we build from `state.pagesData` 
  - also takes a callback that fires when the system asks to pop (eg. when the back button is tapped) which we use to update our app state 
- I removed the routes, so I could easily find where I needed to update the navigation - there's a lot more cleaning up to do but I wanted to make it a separate PR as this was already large

Reply:
sweet thanks for the detailed comment @nickmeinhold!